script/src/syscalls/spawn.rs

```diff
@@ -237,24 +218,128 @@ where
                 return Ok(true);
             }
         }
-        // Check result.
+        // Deduct cycles used to build the child machine
```

Comment:
We need to deduct extra cycles before `load_program`. Consider a scenario: an evil user tries to run a non-RISC-V program, and the syscall spawn will fail at `load_program`. We have initialized the VM instance( includes memory, syscalls, etc. ) for running the program, but the eval user only need to pay 500 cycles.